chain/chaintest/action_test_helpers.go

```diff
@@ -81,3 +82,20 @@ func (test *ActionTest) Run(ctx context.Context, t *testing.T) {
 		}
 	})
 }
+
+// RunBenchmark executes the [ActionTest] and make sure all the benchmark assertions pass.
```

Comment:
We're using this in a different way here ie. ActionTest does not assume that the state will be re-used `b.N` times.

Could we switch to using a separate type here? We may need to switch to a `CreateState` function so that each benchmark has its own state instead of mutating the same one over and over. If a given benchmark can re-use the same state, it can return the same one from that function.